coins/ethereum/contracts/Router.sol

```diff
@@ -0,0 +1,101 @@
+//SPDX-License-Identifier: AGPLv3
+pragma solidity ^0.8.0;
+
+import "./Schnorr.sol";
+
+contract Router is Schnorr {
+    // contract owner
+    address owner;
+
+    // nonce is incremented for each batch of transactions executed
+    uint256 nonce; 
+
+    // prevents re-entrancy
+    uint8 internal locked;
+
+    struct PublicKey {
+        uint8 parity;
```

Comment:
What are the gas savings if this is always even/odd? Asking as Bitcoin already requires a fixed parity (would have to check which). Carrying that to Ethereum wouldn't be an issue, and it saves passing around a word.

Reply:
I suppose we'd save an SSTORE there - could modify to force it to be fixed if you'd like. but then I suppose we'd have to force the group key to always have the right parity somehow?

Reply:
Right. We can offset by one until it has parity, which we're doing for BTC. It's incredibly minor on our end and incredibly minor here as well. I don't care to further discuss it ad infinitum.

Reply:
updated, idk if you had a preference for 27 or 28, I just put 27